### DISCUSSION TITLE
Nextjs / Reactjs support

### DISCUSSION THREAD

**Discussion by hamburger-fries** [skeletonlabs/skeleton#625] [Ideas]
Love your project!

I mostly use [T3 Stack](https://github.com/t3-oss/create-t3-app).
The idea of another cool UI-KIT for Tailwind is very exciting!
Mostly what others do is charge money for a ReactJS pre-built components library.

I am of the strong belief that if the project is just a One-Off and will not involve multiple developers then using any UI-Framework is good. However, if the project involves many people and is ongoing then to me the only option is Tailwind. The main point is 'headless' systems will have more compatibility and will reduce stress for new team members.

I am on a dev team for a high-traffic site. If you end up working more closely with ReactJS I would love to test your stuff in a fast-paced development environment.

**Comment by endigo9740** [comment]
Hey @forestcall, yeah Svelte is and will continue to be our focus for the foreseeable future. At least through the v1.0 launch. However, we have talked about branching out to support other frameworks in the future. There's no technical limitations for this, but it's more a matter of logistics. Currently I'm the only full time developer, but we do have a small handful of dedicated contributors.

That said, technically portions of Skeleton platform agnostic as long asTailwind is present, Next/React included. This, however, will be limited to purely design tokens, base styles, and elements (including buttons, cards, etc) found here:

https://www.skeleton.dev/elements/core

When you need logic, that's where components come in, and for that all ours are written in Svelte. It will take a pretty hefty effort to support even a single other framework. However, Skeleton is open source. If you're interested in contributing and making this happen then let us know. We can discuss further.

Personally I don't have a working knowledge of React, so I would not be the most qualified person to port our components to this. My background before Svelte was in Angular. But I would welcome support from anyone that wished to contribute.

**Comment by sbusso** [comment]
First, very good work on the UI and composition, I like the way Skeleton manages themes (as with primary colors) and base components (as with a btn class), and I'd like to see more UI projects following this semantic. Working on a React/Tailwind project I extracted the theming and styles, and I think it would be beneficial for other frameworks to have this theming and base components available as a Tailwind plugin without Svelte dependency and the more advanced elements. It would also facilitate the development of framework specific components.

**Reply by endigo9740** [reply]
@sbusso It's in the cards for the future. Svelte is, an always will be, the tip of the spear for us. But we do have plans to expand the components to other frameworks in the future.

FYI technically everything under the Tailwind > Elements section is framework agnostic, only needing Tailwind installed. We've not done extensive testing with it, but there's no technical limitations there. If you give it a go and make something worth sharing please let us know!
